### PR TITLE
docs: update go library quick start examples

### PR DESCRIPTION
**Pull request by Wwwsylvia**
The quick start examples for the go library are now available at [pkg.go.dev](https://pkg.go.dev/oras.land/oras-go/v2#example-package-PullFilesFromRemoteRepository).
This PR is to update the go page to reference the latest examples.

Preview page: https://deploy-preview-256--oras-project.netlify.app/docs/client_libraries/go

### CODE REVIEW

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by sajayantony — APPROVED** [review]
Need to make sure this example doesn’t go out of sync with the the library.

File: docs/client_libraries/go.mdx [hunk 1]
```diff
@@ -43,210 +43,13 @@ go mod tidy
...
-    return err
-}
-
-func main() {
-    if err := pushFiles(); err != nil {
-        panic(err)
-    }
-}
-```
+See [example](https://pkg.go.dev/oras.land/oras-go/v2#example-package-PushFilesToRemoteRepository).
```
Comment (SamirPS) [inline docs/client_libraries/go.mdx:46]: i think it's better to say "See this example" no? But else LGTM

**Review by SamirPS — CHANGES_REQUESTED** [review]

File: docs/client_libraries/go.mdx (hunk 1 above)
Comment (TerryHowe) [reply docs/client_libraries/go.mdx:46]: Might be better just to change this to a bullet list, but this is good enough for now

**Review by TerryHowe — APPROVED** [review]

**Comment by TerryHowe** [comment]
I think a bullet list would be better, but this works

**Pull request merged**